geometry_msgs/Pose[] subgoals 